src/FileDownloader/ChunkFileDownloader.cs

```diff
@@ -13,22 +13,29 @@ namespace ProjectCeleste.GameFiles.GameScanner.FileDownloader
     public class ChunkFileDownloader : IFileDownloader
     {
         internal const int MaxChunkSize = 10 * 1024 * 1024; //10Mb
-        private const int MaxConcurrentDownloads = 6;
+        public const int MaxConcurrentDownloads = 10;
 
+        private readonly int _concurrentDownloads;
         private readonly Stopwatch _downloadSpeedStopwatch;
         private readonly string _downloadTempFolder;
-
         private readonly ConcurrentDictionary<long, string> _completedChunks = new ConcurrentDictionary<long, string>();
-        private ConcurrentQueue<FileRange> _chunkDownloadQueue;
 
+        private ConcurrentQueue<FileRange> _chunkDownloadQueue;
         private long _downloadSizeCompleted;
-        private int _activeDownloads = 1;
-        private bool _downloadFailed = false;
 
-        public ChunkFileDownloader(string httpLink, string outputFileName, string tmpFolder)
+        public ChunkFileDownloader(string httpLink, string outputFileName, string tmpFolder, int concurrentDownload = 0)
         {
             DownloadUrl = httpLink;
             FilePath = outputFileName;
+
+            if (concurrentDownload <= 0)
+                concurrentDownload = Environment.ProcessorCount * 5;
+
+            if (concurrentDownload > MaxConcurrentDownloads)
+                concurrentDownload = MaxConcurrentDownloads;
```

Comment:
This code becomes dangerous if we do not have this specific if-statement. See comment above.